### PR TITLE
[Utils] Installer: WasmEdge installer

### PR DESCRIPTION
**Pull request by SAtacker**
- Follows the old installer
- Added basic plugin download on top of old

Signed-off-by: Shreyas Atre <shreyasatre16@gmail.com>

### CODE REVIEW

**Comment by SAtacker** [comment]
Test:
```bash
curl -sSf https://raw.githubusercontent.com/SAtacker/WasmEdge/installer_py_pre_plugin/utils/install.sh | INSTALL_PY_URL=https://raw.githubusercontent.com/SAtacker/WasmEdge/installer_py_pre_plugin/utils/install.py bash -s -- --plugins wasi_crypto:0.11.0 -D
```

File: utils/install.py [hunk 1]
```diff
@@ -0,0 +1,1667 @@
...
+        required=False,
+        default=platform.machine(),
+        help="Machine ex- x86_64, aarch64",
+    )
+    parser.add_argument(
+        "--dist",
+        dest="dist",
+        required=False,
+        default=None,
+        help="Dist ex- ubuntu20.14,manylinux2014",
```
Comment (hydai) [inline utils/install.py:1658]: should be 20.04

File: utils/install.py [hunk 2]
```diff
@@ -0,0 +1,1667 @@
...
+        default=get_latest_github_release("WasmEdge/WasmEdge"),
+        help="GitHub tag for uninstall script",
+    )
+    parser.add_argument(
+        "--plugins",
+        dest="plugins",
+        required=False,
+        default=[],
+        nargs="*",
+        help="Supported Plugins. Example\n"
```
Comment (hydai) [inline utils/install.py:1600]: Please add an experimental option warning, we may change this behavior once the plugin metadata is finished.

**Review by hydai — CHANGES_REQUESTED** [review]
After we introduced the python installer, it would be better to have some linter checks in the workflow for this file, e.g. pylint

**Comment by hydai** [comment]
Also, looks like the uninstaller is failed.

**Comment by SAtacker** [comment]
That's misleading. The installer script does not fail as per the request. It simply shows that the URL is not reachable and doesn't installs.
Hence Uninstaller cannot find the installed path and it fails.

File: utils/install.py (hunk 1 above)
Comment (SAtacker) [reply utils/install.py:1658]: Yeah, thanks

File: utils/install.py (hunk 2 above)
Comment (SAtacker) [reply utils/install.py:1600]: Alright

**Comment by hydai** [comment]
> @hydai That's misleading. The installer script does not fail as per the request. It simply shows that the URL is not reachable and doesn't installs. Hence Uninstaller cannot find the installed path and it fails.

Oh, I see, the error actually failed in the previous steps.

**Review by hydai — APPROVED** [review]
LGTM. @q82419 Could you help me to check this again? If everything goes fine, let's merge this in this release.

**Review by q82419 — APPROVED** [review]

**Comment by q82419** [comment]
The test-install-script CI failed.

**Comment by SAtacker** [comment]
> The test-install-script CI failed.

If it's not any other reason please refer [this](https://github.com/WasmEdge/WasmEdge/pull/1914#issuecomment-1277062276)

**Comment by hydai** [comment]
Will merge this after the python installer workflow is green.

**Comment by SAtacker** [comment]
Sorry, it looks like I forgot to change the debug option for old installer CI

**Pull request merged**